### ISSUE TITLE
LazyInitializationException

### ISSUE THREAD

**Issue by freke1981SWE** [n2cms/n2cms#792]
Hello! Recently we started to get these types of errors. What can cause this?

```
[LazyInitializationException: Initializing[N2.ContentItem#6543]-failed to lazily initialize a collection of role: N2.ContentItem.Details, no session or session was closed]
   NHibernate.Collection.AbstractPersistentCollection.ThrowLazyInitializationException(String message) +199
   NHibernate.Collection.AbstractPersistentCollection.ThrowLazyInitializationExceptionIfNotConnected() +41
   NHibernate.Collection.AbstractPersistentCollection.Initialize(Boolean writing) +35
   NHibernate.Collection.Generic.PersistentGenericBag`1.GetEnumerator() +27
   System.Linq.Enumerable.Any(IEnumerable`1 source, Func`2 predicate) +98
   N2.ContentItem.GetDetail(String detailName) +46
   N2.Persistence.Proxying.<>c__DisplayClass14_0.<GetGetDetail>b__0(IInvocation invocation) +70
   N2.Persistence.Proxying.DetailPropertyInterceptor.Intercept(IInvocation invocation) +83
   Castle.DynamicProxy.AbstractInvocation.Proceed() +147
   Castle.Proxies.ProductNavigationPageProxy.get_StormCategoryID() +167
```
We cache the N2 pages in an InMemory cache, can this cause problem when getting items from the cache?

**Comment by freke1981SWE** [comment]
I found out that the cached items, has lost the NHibernate connection...

**Comment by DejanMilicic** [comment]
N2 pages cannot be kept in cache. I also tried implementing caching and I ended up with a scheme where I find set of pages, then cache their IDs.
After reading cache, I get IDs from cache, and then pull pages with those IDs.

What I did not try, and might be one of the possible solution, would be to serialize Page into JSON, then store JSON, and later on hydrate pages from JSON.... but this is just idea, I did not try it yet.
